### PR TITLE
core/lib: Add macros/utils.h header

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

The macros CONCAT(), MIN(), and MAX() are defined over and over again in RIOT's code base. This de-duplicates the code by moving the macros to a common place.

### Testing procedure

Generated binaries don't change, as this only a de-duplication of macros that doesn't change their definition.

### Issues/PRs references

None

### CODE REVIEW

**Comment by benpicco** [comment]
bors try

**Review by benpicco — APPROVED** [review]
This is a useful addition, please squash

File: tests/bench_sys_base64/main.c [hunk 1]
```diff
@@ -23,10 +23,9 @@
 
 #include "base64.h"
 #include "fmt.h"
+#include "macros/min.h
```
Comment (benpicco) [inline tests/bench_sys_base64/main.c:26]:
```suggestion
#include "macros/min.h"
```

**Comment by maribu** [comment]
All green (even a full Murdock run).

**Comment by aabadie** [comment]
bors merge

**Comment by aabadie** [comment]
bors merge

**Comment by maribu** [comment]
bors retry

Maybe it's time to disable `tests/gnrc_rpl` until the test's flakyness is addressed?

**Comment by maribu** [comment]
Thx :)

**Pull request merged**